### DISCUSSION TITLE
engineering-practice-naming-conventions

### DISCUSSION THREAD

**Discussion by giscus** [WorkSafeBC-Common-Engineering/Common.Engineering.Discussion#5] [Technical Blog]
# engineering-practice-naming-conventions

https://wsbctechnicalblog.github.io/engineering-practice-naming-conventions.html

**Comment by wpschaub** [comment]
Martin, which is your favourite case convention and why?

**Reply by MerlinDarkstar** [reply]
Great question - I've used and experimented with all of them, and sometime a choice was forced upon me.  The one I found most effective and easy to use is camelCase for member variables, instances and parameters, and PascalCase for class names and types.  

One notable wrinkle - if a private variable whose scope is the entire class (or a private property), I like to preface it with an underscore "_" to highlight its' wide scope nature; this is a relatively rare occurrence, however.

**Reply by MerlinDarkstar** [reply]
A couple of other points to note: the dash "-" can be interpreted as a math operator in some environments (e.g. HTML), so I tend to be hesitant and very cautious around variables or class names that contain them.
Using a date convention that has decreasing scope left to right (e.g. YYYMMDDHHMMSSmm) is most handy and makes searches based on dates consistent and straight forward.
